src/test/ui/async-await/issue-73050.rs

```diff
@@ -0,0 +1,12 @@
+// check-pass
+// edition:2018
+
+#[allow(unused)]
+async fn foo<'a>() {
+    let _data = &mut [0u8; { 1 + 4 }];
+    bar().await
+}
```

Comment:
Weird that this would need `async`, presumably `fn foo<'a: 'a>` (forcing lifetime parameters aka "early-bound" as opposed to only the signature being lifetime-polymorphic aka "late-bound) would also work?

Reply:
I can't get it to ICE: [playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=69d4cae92bffe277cb7791d01c7b233c)

The ICE happened when unifying the generator witness with the generator type (IIRC?), so maybe there's something special about that operation.